pkg/app/server.go

```diff
@@ -400,8 +402,23 @@ func buildMetricsServer(m *metricshandler.MetricsHandler, durationObserver prome
 	mux.Handle("/debug/pprof/symbol", http.HandlerFunc(pprof.Symbol))
 	mux.Handle("/debug/pprof/trace", http.HandlerFunc(pprof.Trace))
 
+	// Add metricsPath
 	mux.Handle(metricsPath, promhttp.InstrumentHandlerDuration(durationObserver, m))
 
+	// Add livezPath
+	mux.Handle(livezPath, http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+
+		// Query the Kube API to make sure we are not affected by a network outage.
+		got := client.CoreV1().RESTClient().Get().AbsPath("/apis/").Do(context.Background())
```

Comment:
Should we query Kubernetes' API readyz endpoint https://kubernetes.io/docs/reference/using-api/health-checks/ instead to detect an outage of the kubernetes apiserver?

Reply:
+1, API (`/apis`) discoverability used here is a subset of `/livez`. I'll make the changes.

Reply:
Should this be `/livez` or rather `/readyz` ? 

The Kubernetes API Server could be healthy (/livez = true) but not be able to accept client request (/ready = false).

Reply:
`/livez` knows when to restart the container, and thus knows if there's been an outage. Besides, it makes sense to point the our `/livez` to the cluster's `/livez` to ensure the same thing.

`readinessProbe` is currently set to the telemetry metrics' availability in our [`jsonnet` config](https://github.com/kubernetes/kube-state-metrics/blob/6de105ebbe0eeb5d97d9e6adf1cc83314434e8cc/jsonnet/kube-state-metrics/kube-state-metrics.libsonnet#L198-L201), which seems okay. A more robust approach would be coupling that with the cluster's `/readyz` and exposing that under a dedicated `/readyz` (we don't have a dedicated endpoint just yet). This would mean that (a) the cluster components are ready, and (b) the binary itself is ready. I can open another PR for that.

Reply:
Sounds good to me! Let's open another PR for that

Reply:
One question: We already have a /healthz path, what would this /livez path then be used for?

Reply:
`/healthz` will send out a `200` if the binary is running, `/readyz` will send out a `200` if the exposition machinery is working as expected (we are ready to serve metrics), and `/livez` will send out a `200` if none of the collectors are affected by any outages (collectors depend on the Kube API to gather data).

A `/healthz` endpoint would be more suitable for a `startupProbe`, which is especially useful if we believe the binary takes a considerable time to start.

Reply:
Thanks for clarifying! Could you add this to the README and update our jsonnet as well?